DICOM/src/Microsoft.Health.Dicom.Anonymizer.CommandLineTool/AnonymizerLogic.cs

```diff
@@ -0,0 +1,90 @@
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Dicom;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Tool
+{
+    internal static class AnonymizerLogic
+    {
+        internal static async Task AnonymizeAsync(AnonymizerOptions options)
+        {
+            var engine = new AnonymizerEngine(
+                    options.ConfigurationFilePath,
+                    new AnonymizerEngineOptions(options.ValidateInput, options.ValidateOutput));
+            if (options.InputFile != null && options.OutputFile != null)
+            {
+                if (IsSamePath(options.InputFile, options.OutputFile))
+                {
+                    Console.Error.WriteLine("Input and output file path are the same! Please check file names.");
+                }
+
+                await AnonymizeOneFileAsync(options.InputFile, options.OutputFile, engine, options.SkipFailedItem);
+            }
+            else if (options.InputFolder != null && options.OutputFolder != null)
+            {
+                if (IsSamePath(options.InputFolder, options.OutputFolder))
+                {
+                    Console.Error.WriteLine("Input and output folders are the same! Please choose another folder.");
+                }
+
+                foreach (string file in Directory.EnumerateFiles(options.InputFolder, "*.dcm", SearchOption.AllDirectories))
+                {
+                    await AnonymizeOneFileAsync(file, Path.Join(options.OutputFolder, Path.GetRelativePath(options.InputFolder, file)), engine, options.SkipFailedItem);
+                }
+
+                Console.WriteLine("Anonymization finished!");
+            }
+            else
+            {
+                Console.Error.WriteLine("Invalid command line. Please specify inputFile( or inputFolder) and outputFile( or outputFolder) at the same time.");
+            }
+        }
+
+        private static bool IsSamePath(string inputPath, string outputPath)
+        {
+            string inputFullPath = Path.GetFullPath(inputPath)
+                .TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+            string outputFullPath = Path.GetFullPath(outputPath)
+                .TrimEnd(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
+
+            return string.Equals(inputFullPath, outputFullPath, StringComparison.InvariantCultureIgnoreCase);
+        }
+
+        internal static async Task AnonymizeOneFileAsync(string inputFile, string outputFile, AnonymizerEngine engine, bool skipFailedItem)
+        {
+            try
+            {
+                DicomFile dicomFile = await DicomFile.OpenAsync(inputFile).ConfigureAwait(false);
+                engine.AnonymizeDataset(dicomFile.Dataset);
+
+                if (!string.IsNullOrEmpty(Path.GetDirectoryName(outputFile)))
```

Comment:
>if (!string.IsNullOrEmpty(Path.GetDirectoryName(outputFile))) [](http://example.com/codeflow?start=16&length=61)

you can make the code simpler by remove this since even this value is empty, you has try to catch it. :grinning:

Reply:
> you can make the code simpler by remove this since even this value is empty, you has try to catch it. 😀
No, you're wrong.
In case the parent folder name is empty, we don't need to create the folder, but still need to save the output file.
If exception happens, the save step will be skipped.

Reply:
well, after some hot dicusstions, I think you can ignore my original comment, although I am still not fan of this code

Reply:
If outputFile is a file name without directory, Path.GetDirectoryName(outputFile) will return empty string and Directory.CreateDirectory will throw exception. But we still need to save this file.